Inedo.DependencyScan/NpmDependencyScanner.cs

```diff
@@ -22,13 +22,41 @@ public override async Task<IReadOnlyCollection<ScannedProject>> ResolveDependenc
 
         private static IEnumerable<DependencyPackage> ReadDependencies(JsonDocument doc)
         {
-            if (!doc.RootElement.TryGetProperty("dependencies", out var dependencies))
+            if (!doc.RootElement.TryGetProperty("lockfileVersion", out var lockFileVersion))
                 yield break;
 
-            foreach (var d in dependencies.EnumerateObject())
+            var lockFileVersionString = lockFileVersion.ToString();
+
+            JsonElement npmDependencyPackages;
+            if (lockFileVersionString.Equals("3"))
+            {
+                if (!doc.RootElement.TryGetProperty("packages", out npmDependencyPackages))
+                    yield break;
+            }
+            else 
+            {
+                if (!doc.RootElement.TryGetProperty("dependencies", out npmDependencyPackages))
+                    yield break;
+            }
+
+            foreach (var npmDependencyPackage in npmDependencyPackages.EnumerateObject())
             {
-                string name = d.Name;
-                var version = d.Value.GetProperty("version").GetString();
+                string name = string.Empty;
+                if (lockFileVersionString.Equals("3"))
+                {
+                    // skip the self reference package
+                    if (npmDependencyPackage.Name.Equals(string.Empty))
+                        continue;
+
+                    // drop node_modules/ prepended string
+                    name = npmDependencyPackage.Name.Replace("node_modules/", string.Empty);
```

Comment:
Perhaps a more generous / general substring should be used here as this is the path from project root, which may not just be `node_modules/package-name`. Perhaps `Substring(Name.LastIndexOf('/'), Name.Length)` or similar?

Reply:
`npmDependencyPackage.Name` can contain severeal "node_modules/" elements and package names in cases of transitive dependencies. How about the following code instead:

```suggestion
                    var lidx = npmDependencyPackage.Name.LastIndexOf("node_modules/") + 13;
                    if (lidx < 13 || lidx >= npmDependencyPackage.Name.Length)
                        name = npmDependencyPackage.Name;
                    else
                        name = npmDependencyPackage.Name.Substring(lidx);
```

Reply:
@szimmer-dap per your suggestion, I've incorporated your comments in this commit: https://github.com/Inedo/pgscan/pull/35/commits/ed5c6785fb285b507e6da1b01828dc2d313e10b6